fabric-screen-api-v1/src/client/java/net/fabricmc/fabric/api/client/screen/v1/LevelScreenProviderRegistry.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.screen.v1;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import net.minecraft.client.gui.screen.world.LevelScreenProvider;
+import net.minecraft.registry.RegistryKey;
+import net.minecraft.world.gen.WorldPreset;
+
+import net.fabricmc.fabric.mixin.screen.LevelScreenProviderAccessor;
+
+/**
+ * Adds registration hooks for {@link LevelScreenProvider}s.
+ */
+public final class LevelScreenProviderRegistry {
```

Comment:
We generally use interfaces for most public API like this, I would move the logic (with the logger) to an impl class.